### PR TITLE
Chat: Display Enhanced Context settings on first chat

### PR DESCRIPTION
**Pull request by abeatrix**
CLOSE https://github.com/sourcegraph/cody/issues/2839

image

- As discussed with @toolmantim in [this figma comment thread](https://www.figma.com/file/jEsnzgsf0hNuJbqB9pKaGm?type=design&node-id=1447-38603&mode=design#739536448), we will open Enhanced Settings on first chat by default.
- Add dropdown button for Enhanced Context Settings
- Existing button for Enhanced Context Settings is now for toggling enhanced context enablement.

## Test plan

- Updated e2e tests

To test this manually:
1. Open a new Cody Chat
2. The Enhanced Context settings should be opened by default
3. Send a new message
4. the Enhanced Context settings windows is closed when you move focus to the chat input box

### After

First chat for Dot Com users:

image

First chat for enterprise users:

image

### CODE REVIEW

**Comment by abeatrix** [comment]
@toolmantim before I fix the e2e tests to reflect the new behavior, can you take a look and let me know if this aligns with what you had in mind?

**Comment by toolmantim** [comment]
@abeatrix Nice! Yeah checking it out now. Quick one: I spotted `CodyVSCodeExtension:useEnhancedContextToggler:clicked` event isn't triggered when you click it via the new button.

**Comment by abeatrix** [comment]
> @abeatrix Nice! Yeah checking it out now. Quick one: I spotted `CodyVSCodeExtension:useEnhancedContextToggler:clicked` event isn't triggered when you click it via the new button. (edit: we may want to include a metadata property of the source, so we can tell where they click it from — popover or input button)

added a new event `useEnhancedContextInputToggler` event for the new icon click

**Comment by toolmantim** [comment]
Just pushed a bunch of style fixes, and the only thing left I think is some of the keyboard accessibility fixes:
* Tabbing to enhanced context indicator button and hitting space/enter shouldn't refocus the input, that should only happen when context indicator button is clicked. Pressing escape while the enhanced context indicator button is focused should focus the message input.
* If you shift-tab from the popover checkbox back into the message input, the popover should be dismissed

**Review by toolmantim — APPROVED** [review]
Approving to unblock, given how close this is

**Comment by abeatrix** [comment]
> If you shift-tab from the popover checkbox back into the message input, the popover should be dismissed
@toolmantim I couldn't get this to work, but I will work on it in a follow-up as this is an existing behavior. For now, users can hit Escape when the input box is focused though to dismiss the popover.

File: vscode/webviews/Chat.tsx [hunk 1]
```diff
@@ -517,8 +526,9 @@ export const Chat: React.FunctionComponent<React.PropsWithChildren<ChatboxProps>
                         <div className={styles.contextButton}>
                             <EnhancedContextSettings
                                 isOpen={isEnhancedContextOpen}
-                                setOpen={setIsEnhancedContextOpen}
+                                setOpen={onEnhancedContextTogglerClick}
                                 presentationMode={userInfo.isDotComUser ? 'consumer' : 'enterprise'}
+                                isFirstChat={transcript.length < 1}
```
Comment (toolmantim) [inline vscode/webviews/Chat.tsx:531]: I only spotted this logic. This is more like `isEmptyChat` which isn't what we want here. We want it to appear on the first chat that's created for first time users, not every time a chat is created.

**Pull request merged**